service/cloudfront/sign/sign_cookie_test.go

```diff
@@ -0,0 +1,85 @@
+package sign
+
+import (
+	"crypto/rsa"
+	"testing"
+	"time"
+)
+
+const (
+	policyName    = "CloudFront-Policy"
+	signatureName = "CloudFront-Signature"
+	keyName       = "CloudFront-Key-Pair-Id"
+)
+
+func TestSignCookie(t *testing.T) {
+	privKey, err := rsa.GenerateKey(randReader, 1024)
```

Comment:
Instead of using randReader directly in tests, this should be replaced with the test rand reader so predictable signatures can be tested similar to `TestSignURL`

```go
	origRandReader := randReader
	randReader = newRandomReader(rand.New(rand.NewSource(1)))
	defer func() {
		randReader = origRandReader
	}()
```